lib/src/react_client/factory_util.dart

```diff
@@ -63,24 +72,32 @@ void convertRefValue(Map args) {
   }
 }
 
-void convertRefValue2(Map args, {bool convertCallbackRefValue = true}) {
-  var ref = args['ref'];
-
-  if (ref is Ref) {
-    args['ref'] = ref.jsRef;
-    // If the ref is a callback, pass ReactJS a function that will call it
-    // with the Dart Component instance, not the ReactComponent instance.
-    //
-    // Use CallbackRef<Null> to check arity, since parameters could be non-dynamic, and thus
-    // would fail the `is CallbackRef<dynamic>` check.
-    // See https://github.com/dart-lang/sdk/issues/34593 for more information on arity checks.
-  } else if (ref is CallbackRef<Null> && convertCallbackRefValue) {
-    args['ref'] = allowInterop((dynamic instance) {
-      // Call as dynamic to perform dynamic dispatch, since we can't cast to CallbackRef<dynamic>,
-      // and since calling with non-null values will fail at runtime due to the CallbackRef<Null> typing.
-      if (instance is ReactComponent && instance.dartComponent != null) return (ref as dynamic)(instance.dartComponent);
-      return (ref as dynamic)(instance);
-    });
+void convertRefValue2(
+  Map args, {
+  bool convertCallbackRefValue = true,
+  List<String> additionalRefPropKeys = const [],
+}) {
+  final refKeys = ['ref', ...additionalRefPropKeys];
+
+  for (var refKey in refKeys) {
+    var ref = args[refKey];
+    if (ref is Ref) {
+      args[refKey] = ref.jsRef;
+      // If the ref is a callback, pass ReactJS a function that will call it
+      // with the Dart Component instance, not the ReactComponent instance.
+      //
+      // Use _CallbackRef<Null> to check arity, since parameters could be non-dynamic, and thus
+      // would fail the `is _CallbackRef<dynamic>` check.
+      // See https://github.com/dart-lang/sdk/issues/34593 for more information on arity checks.
+    } else if (ref is CallbackRef<Null> && convertCallbackRefValue) {
+      args[refKey] = allowInterop((dynamic instance) {
+        // Call as dynamic to perform dynamic dispatch, since we can't cast to _CallbackRef<dynamic>,
+        // and since calling with non-null values will fail at runtime due to the _CallbackRef<Null> typing.
+        if (instance is ReactComponent && instance.dartComponent != null)
+          return (ref as dynamic)(instance.dartComponent);
```

Comment:
#nit [prefer using curly braces for multiline if statement](https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures) (probably just got wrapped during formatting)